apps/aecore/src/aec_blocks.erl

```diff
@@ -292,32 +293,43 @@ hash_internal_representation(B = #block{}) ->
     aec_headers:hash_header(to_header(B)).
 
 
-%% TODO: implement validation of microblocks
--spec validate(block(), binary()) -> ok | {error, term()}.
-validate(Block, _LeaderKey) ->
+validate_key_block(Block) ->
+    case aec_headers:validate_key_block_header(to_header(Block)) of
+        ok ->
+            Validators = [fun validate_no_txs/1,
```

Comment:
I would expect the txs absence to be checked during serialization, by not having txs in the key block structure?

Reply:
Yes, this should go away after @jur0 's refactoring?

Reply:
Correct, txs won't be there anymore in key blocks.

Reply:
@lucafavatella @jur0 Since this PR should go in before we have big @jur0 's refactoring I wanted this check to be here so that we don't forget. Objections?

Reply:
> Objections?

No.